### PR TITLE
docs: Add "Keeping Branch Up to Date and Resolving Merge Conflicts" section

### LINKED ISSUE #118: Feature: Add a Resolve Merge Conflicts section on README
### Type of feature

🍕 Feature

### Current behavior

The process of adding contributors to this repository gives high potential of resolving conflicts before PR can be merged. It happens because new contributors will always add themselves at the end of the contributors array.

### Suggested solution

- Add a section in the README to walk through the process of resolving merge conflicts.
- Add screeshots/screen recordings (GIF)/video walkthrough for better understanding.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by adiati98**
## Description

This PR adds the Keeping Branch Up to Date and Resolving Merge Conflicts section to walk through the process for this project.

## Note for Reviewers

- Because there is CLI involved for this project, screen recordings (GIFs) will be more clear than screenshots. So there will be GIFs (or video walkthrough) provided, but how to add them here is to be advised.

-  One of the steps to resolve conflict is to move contributor's profile details to the last list in the contributors array. This step is necessary because without it, the PRs that are awaiting for review have to update their branch locally and run `npm run contributors:generate` to regenerate the badge. It will be too complicated and discouraging for this simple project --> this is one of the reasons that the process won't be too clear with screenshots.

- Once this PR is merged, there will be another PR following to add GIFs.

- Videos walkthrough are planned to replace GIFs in the future.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Refers to #118 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

cross fingers GIF

### CODE REVIEW

File: README.md
Comment (CBID2) [inline README.md:115]: [suggestion] It's a good start, but it could be a bit more casual and clearer. Remember, most of the contributors are beginners, so we need to convey this information in an easy-to-read format. 
```suggestion
Merge conflicts are something you'll encounter a lot when contributing to an open source project. They usually occur when there are changes on the same line(s) in the same file(s) from two different branches.
```

File: README.md
Comment (CBID2) [inline README.md:119]:
```suggestion
 

Since the main purpose of the guestbook to add your name to  `.all-contributorsrc` and the `README.md` files, there is a high chance that you will encounter merge conflicts that you need to resolve before your pull request (PR) can be merged. It happens when maintainers have merged PRs before yours while you're working on your changes or waiting for your PR to be reviewed.
```

File: README.md
Comment (CBID2) [inline README.md:131]: [suggestion] I think it would be beneficial to add screenshots that convey each step. It'll help people who are new to open source and GitHub gain a better understanding of this process. To help you, I added comments on where to put them. 
```suggestion
1. Go to your forked repository on GitHub.
{/* add screenshot here */}
2. Click the "Sync fork" button.
{/* add screenshot here */}
3. Click the green "Update branch" button.
{/* add screenshot here */}
```

File: README.md
Comment (CBID2) [inline README.md:163]: [suggestion] Again, adding screenshots would help convey the steps more clearly. 
```suggestion
1. In the `.all-contributorsrc` file:
   - Click the "Accept Both Change" option on the top of your workspace in VSCode.
   {/* add screenshot here */}
   - Move your profile details to the end of the contributors array and fix anything necessary.
   
   {/* add screenshot here */}

2. In the `README.md` file:

   - Click the "Accept Incoming Change" option on the top of your workspace in VSCode for each conflict in this file.

3. Run `npm run contributors:generate`.

   You will now see that the all contributors badge has been incremented, and your profile is generated at the end of the contributors list in the `README.md` file.
   
   {/* add screenshot here */}

```

File: README.md
Comment (CBID2) [inline README.md:178]: [suggestion] This seems best suited as a standalone paragraph instead of a heading. Also, mention the pizza verse repository so that new contributors can continue to grow their skills. 
```suggestion
Congratulations on your first contribution! 🎉 You can add this contribution to your [open source](https://opensauced.pizza) resume and profile as [outlined in the course](https://github.com/open-sauced/intro/blob/main/05-how-to-contribute-to-open-source.md)!
Hungry for more contributions? Check out our [pizza verse](https://github.com/open-sauced/pizza) repository and join us in celebrating pizza! 😄 🍕 
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @adiati98! :) The guide looks pretty good so far! :) I just gave some feedback in my inline comments. I hope you find them helpful! :)

File: README.md
Comment (adiati98) [reply README.md:131]: As I mentioned before, I have a screen recorder (GIF) for visualization. But I don't know how I add them here right now.

I'm waiting until this PR is merged so I can drag and drop them later directly on GitHub :)

File: README.md
Comment (adiati98) [reply README.md:178]: It was a standalone paragraph. But I added the title to make it noticable so folks won't missed it because we have much longer "Getting Started" section.

I believe we still need to make this paragraph noticable. Would an `<hr>` after the "Resolving Conflicts" section work better here than a title?

File: README.md
Comment (adiati98) [reply README.md:115]: In that case, I rather use the word "common" than "a lot".
For beginners in open source, merge conflicts are something terrifying. So reading the word "a lot" can discourage them if they don't have smooth first experience.

How about: 

Merge conflicts are something you'll *commonly* encounter when contributing to an open source project. They usually occur when there are changes on the same line(s) in the same file(s) from two different branches.

**Comment by adiati98** [comment]
> Hi @adiati98! :) The guide looks pretty good so far! :) I just gave some feedback in my inline comments. I hope you find them helpful! :)

Thanks, @CBID2!
I left some comments and also added "Note for Reviewers" section in the PR description :)

File: README.md
Comment (adiati98) [inline README.md:114]: I fixed this and used the word "commonly" here :)

File: README.md
Comment (adiati98) [inline README.md:116]: I have fixed this as well :)

File: README.md
Comment (adiati98) [inline README.md:179]: I finally came up with this idea for the title, so folks won't miss it. :)
Any thoughts?

**Review by adiati98 — COMMENTED** [review]
@CBID2 I made some fixes for you to check out. Thank you!😊

**Review by CBID2 — APPROVED** [review]
Looks good to me! :)

File: README.md
Comment (CBID2) [reply README.md:179]: I like @adiati98.

**Review by CBID2 — CHANGES_REQUESTED** [review]
You got some merge conflicts on here that need solving @adiati98

**Comment by adiati98** [comment]
> You got some merge conflicts on here that need solving @adiati98

How ironic is that, @CBID2, because this PR is about resolving conflicts 😂

**Comment by adiati98** [comment]
> Looks good to me! :)

Thanks! Now just need to discuss about the screen recordings before I make this ready for review ;)

**Review by BekahHW — APPROVED** [review]
Although I think that adding screenshots are helpful, they aren't necessary especially if this remains in the readme. With the README, it's important that we don't overwhelm contributors with too much. I think that screenshots will start to make the README feel too long. I do have this [blog post ](https://dev.to/opensauced/keeping-your-branch-up-to-date-and-handling-merge-conflicts-while-waiting-for-pr-reviews-3b3h) written about it that can be referenced here that has screenshots.

Depending on how things change here, we might want to add another folder in the repo for community-resources that includes translations and the info here, but I think for now, the repo is still manageable.

**Comment by adiati98** [comment]
@BekahHW this PR can't be merged because there were unresolved reviews from @CBID2. Now I closed them as resolved. You can try again.

If it doesn't work, you need to override @CBID2 review. I did this before in one PR, but forgot to take note.
I think there's an option to do so.
And at the end, you will get something like this:

dismissed stale

**Review by CBID2 — APPROVED** [review]
Looks great now @adiati98! :) Sorry for taking so long. I thought I approved your PR. I guess I did in my head lol 😂

**Pull request merged**
